### PR TITLE
Return SDK errors from Metadata module

### PR DESCRIPTION
**Pull request by llama-del-rey**
## Description

closes: #978 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/metadata/keeper/msg_server.go [hunk 1]
```diff
@@ -106,11 +104,11 @@ func (k msgServer) DeleteScopeDataAccess(
 
 	existing, found := k.GetScope(ctx, msg.ScopeId)
 	if !found {
-		return nil, fmt.Errorf("scope not found with id %s", msg.ScopeId)
+		return nil, sdkerrors.ErrInvalidRequest.Wrapf("scope not found with id %s", msg.ScopeId)
```
Comment (nullpointer0x00) [inline x/metadata/keeper/msg_server.go:107]: Should keep not found constant with how you are doing it above with `sdkerrors.ErrNotFound.Wrapf`

File: x/metadata/keeper/msg_server.go [hunk 2]
```diff
@@ -129,12 +127,12 @@ func (k msgServer) AddScopeOwner(
...
 
 	if err := msg.ValidateBasic(); err != nil {
-		return nil, err
+		return nil, sdkerrors.ErrInvalidRequest.Wrap(err.Error())
 	}
 
 	existing, found := k.GetScope(ctx, msg.ScopeId)
 	if !found {
-		return nil, fmt.Errorf("scope not found with id %s", msg.ScopeId)
+		return nil, sdkerrors.ErrInvalidRequest.Wrapf("scope not found with id %s", msg.ScopeId)
```
Comment (nullpointer0x00) [inline x/metadata/keeper/msg_server.go:135]: Same here: `sdkerrors.ErrNotFound.Wrapf`

File: x/metadata/keeper/msg_server.go [hunk 3]
```diff
@@ -161,12 +159,12 @@ func (k msgServer) DeleteScopeOwner(
...
```
Comment (nullpointer0x00) [inline x/metadata/keeper/msg_server.go:167]: Same here: `sdkerrors.ErrNotFound.Wrapf`

File: x/metadata/keeper/query_server.go [hunk 4]
```diff
@@ -409,25 +411,25 @@ func (k Keeper) Records(c context.Context, req *types.RecordsRequest) (*types.Re
...
-			return &retval, status.Error(codes.InvalidArgument, "a scope or session is required to look up records by name")
+			return &retval, sdkerrors.ErrInvalidRequest.Wrap("a scope or session is required to look up records by name")
 		}
 		// We know that scopeAddr is legit, and that we have a name. So this won't give an error.
 		recordAddr2 := scopeAddr.MustGetAsRecordAddress(req.Name)
 		if recordAddr.Empty() {
 			recordAddr = recordAddr2
 		} else if !recordAddr.Equals(recordAddr2) {
-			return &retval, status.Errorf(codes.InvalidArgument, "record %s does not have name %s", recordAddr, req.Name)
+			return &retval, sdkerrors.ErrNotFound.Wrapf("record %s does not have name %s", recordAddr, req.Name)
```
Comment (nullpointer0x00) [inline x/metadata/keeper/query_server.go:432]: Should this be: `sdkerrors.ErrInvalidRequest`. I see you are using Invalid request in other areas in the query server for checks like this that fail?  Or should those others be changed? Example: ```		recordAddr2 := scopeAddr.MustGetAsRecordAddress(req.RecordName)
		if recordAddr.Empty() {
			recordAddr = recordAddr2
		} else if !recordAddr.Equals(recordAddr2) {
			return &retval, sdkerrors.ErrNotFound.Wrapf("record %s does not have name %s", recordAddr, req.RecordName)
		} ```

**Review by nullpointer0x00 — CHANGES_REQUESTED** [review]
I had a few comments.  Everything else looks good.

**Comment by SpicyLemon** [comment]
I've copied this branch into our repo: [llama/wrap-into-sdkerrors](https://github.com/provenance-io/provenance/compare/llama/wrap-into-sdkerrors). At some point, we can pick this back up and get it across the finish line. For now, I'm going to close this PR.

**Pull request closed without merging**